File: core_arch/src/main/java/com/huyingbao/core/arch/model/RxError.java
package com.huyingbao.core.arch.model;

import androidx.annotation.NonNull;

/**
 * @author liujunfeng
 * @date 2019/1/1
 */
public class RxError extends RxEvent {
    private final Throwable mThrowable;

    private RxError(@NonNull String tag, Throwable throwable) {
        super(tag);
        mThrowable = throwable;
    }

    public static RxError newInstance(@NonNull RxEvent rxEvent, Throwable throwable) {
        return new RxError(rxEvent.getTag(), throwable);
    }

    public Throwable getThrowable() {
        return mThrowable;
    }
}

File: core_arch_kotlin/src/main/java/com/huyingbao/core/arch/model/RxError.kt
package com.huyingbao.core.arch.model

/**
 * Created by liujunfeng on 2019/1/1.
 */
class RxError private constructor(tag: String, val throwable: Throwable) : RxEvent(tag) {
    companion object {

        fun newRxError(tag: String, throwable: Throwable): RxError {
            return RxError(tag, throwable)
        }
    }
}
